llvm_util/llvm2alive.cpp

```diff
@@ -332,9 +332,13 @@ class llvm2alive_ : public llvm::InstVisitor<llvm2alive_, unique_ptr<Instr>> {
       if (!op)
         return error(i);
 
-      // TODO: support for struct indexing
-      if (I.getStructTypeOrNull())
-        return error(i);
+      if (auto structTy = I.getStructTypeOrNull()) {
+        llvm::Value *vals[] = { llvm::ConstantInt::getFalse(i.getContext()), I.getOperand() };
+        auto offset = DL().getIndexedOffsetInType(structTy, llvm::makeArrayRef(vals, 2));
+        // FIXME: size bits shouldn't be a constant
```

Comment:
you can leave as 64 bits, since that's the return value of getIndexedOffsetInType() anyway. It will be truncated/extended as needed.

Reply:
I am not sure I follow. I am not doing anything to `offset`. The FIXME comment is for the argument to `make_intconst` which creates `1` of width 64.

Reply:
Right, I mean: the comment can go away. I think it's fine to leave size fixed with 64 bits.